### PR TITLE
Move Heading and conversions to Location module

### PR DESCRIPTION
**Pull request by xsebek**
- remove the useless `DirInfo` type
- move the `Direction -> Heading` functions to the `Location` module
- split off from #1069
- part of #1043
- part of #707

### CODE REVIEW

File: src/Swarm/Game/Location.hs [hunk 1]
```diff
@@ -71,11 +90,68 @@ deriving instance FromJSON (V2 Int32)
...
+  DRelative e -> case e of
+    DLeft -> relative perp
+    DRight -> relative (fmap negate . perp)
+    DBack -> relative (fmap negate)
+    DDown -> relative (const down)
+    DForward -> relative id
+  DAbsolute e -> cardinal $ toHeading e
+ where
+  cardinal = const
+  relative = id
```
Comment (byorgey) [inline src/Swarm/Game/Location.hs:134]: What is the point of defining `relative = id`?  Why not just have `DLeft -> perp` and so on?  (Maybe this code was already like that but I didn't notice before.)

File: src/Swarm/Language/Syntax.hs [hunk 2]
```diff
@@ -145,93 +133,21 @@ data RelativeDir = DLeft | DRight | DBack | DForward | DDown
...
- where
-  -- name is generate from Direction data constuctor
-  -- e.g. DLeft becomes "left"
-  directionSyntax = toLower . T.tail . from $ case d of
-    DAbsolute x -> show x
-    DRelative x -> show x
-
-  cardinal = DirInfo directionSyntax . const
-  relative = DirInfo directionSyntax
+-- | Direction name is generate from Direction data constuctor
```
Comment (byorgey) [inline src/Swarm/Language/Syntax.hs:136]:
```suggestion
-- | Direction name is generated from Direction data constuctor
```

**Review by byorgey — APPROVED** [review]

File: src/Swarm/Game/Location.hs [hunk 3]
```diff
@@ -71,11 +90,68 @@ deriving instance FromJSON (V2 Int32)
...
+down = zero
+
+-- | The 'applyTurn' function gives the meaning of each 'Direction' by
+--   turning relative to the given heading or by turning to an absolute
+--   heading
+applyTurn :: Direction -> Heading -> Heading
+applyTurn d = case d of
+  DRelative e -> case e of
+    DLeft -> relative perp
+    DRight -> relative (fmap negate . perp)
```
Comment (kostmo) [inline src/Swarm/Game/Location.hs:127]: I've been meaning to replace `fmap negate` with just [`negated`](https://hackage.haskell.org/package/linear-1.22/docs/Linear-Vector.html#v:negated)

**Review by kostmo — APPROVED** [review]

File: src/Swarm/Game/Location.hs (hunk 3 above)
Comment (xsebek) [reply src/Swarm/Game/Location.hs]: Neat, thanks! :+1:

File: src/Swarm/Game/Location.hs (hunk 1 above)
Comment (xsebek) [reply src/Swarm/Game/Location.hs]: No good reason, I think we originally used it for the absolute/relative distinction, but that has disappeared with the direction constructors.

**Pull request merged**